databricks/koalas/indexes.py

```diff
@@ -471,6 +472,41 @@ def any(self, *args, **kwargs):
     def all(self, *args, **kwargs):
         raise TypeError("cannot perform all with this index type: MultiIndex")
 
+    @property
+    def codes(self):
+        """
+        Integers for each level designating which label at each location.
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame({"a": [1, 2, 3]}, index=[list('abc'), list('ddf')])
+        >>> kdf.index.codes
+        [[0, 1, 2], [0, 0, 1]]
+
+        >>> kdf = ks.DataFrame({"a": [1, 2, 3]}, index=[list('bbc'), list('fee')])
+        >>> kdf.index.codes
+        [[0, 0, 1], [1, 0, 0]]
+
+        >>> kdf = ks.DataFrame({"a": [1, 2], "b": [3, 4]}, index=[list('cc'), list('eb')])
+        >>> kdf.index.codes
+        [[0, 0], [1, 0]]
+        """
+        sdf = self._kdf._sdf
+        cols = self._kdf._internal.index_columns
+
+        # adding underscore is to avoid naming conflict with data columns
+        sdf = sdf.withColumn("__order__", F.monotonically_increasing_id())
+        i = 0
+        for col in cols:
+            df = sdf.select(col).distinct()
+            w = Window.orderBy(col)
+            df = df.withColumn("__code_{}".format(str(i)), F.row_number().over(w) - 1)
+            sdf = sdf.join(df, sdf[col] == df[col]).drop(df[col])
+            i += 1
+
+        sdf = sdf.orderBy("__order__").select(sdf.colRegex("`__code_.`"))
+        return DataFrame(_InternalFrame(sdf=sdf)).astype('int').to_numpy().T.tolist()
```

Comment:
Hmmmmm... this collects everything into driver side which can be against [the design principles](https://koalas.readthedocs.io/en/latest/development/design.html#return-koalas-data-structure-for-big-data-and-pandas-data-structure-for-small-data) ... I think we might have to rather explicitly don't support, and let users call `to_pandas()` directly .. ? cc @ueshin

Reply:
ah, in that case, we might also rather explicitly not support `levels`.

Reply:
emm, I see. Then if so, what do you suggest to do next? @ueshin @HyukjinKwon

Reply:
In my work experience, I think most of the time, those method/property related to `index` are used on columns, and pandas should be enough.

Reply:
oh, sorry. @charlesdong1991 and @ueshin. I was confused. it's mapped to levels
Sure, we should add this then. I thought it's mapped to the actual data.

Reply:
It requires to collect all data into driver so we shouldn't implement. @charlesdong1991 do you use this property and `levels` often? If not, let's explicitly don't support. You can refer https://github.com/databricks/koalas/pull/744 to explicitly don't support.

Reply:
sorry for my late reply, didn't have time these days @HyukjinKwon 

tbh, i barely use this `levels` as well as `codes`often, but this can be different across different people and work. I will also drop supports for `levels` as well